tests/integration/operator/operator_test.go

```diff
@@ -253,6 +255,22 @@ spec:
 	})
 })
 
+func getVersionFromIstiodLogs() (string, error) {
```

Comment:
can't we just run something like `c -n istio-system exec istiod-76569d4f64-st6fs -- pilot-discovery version` and parse the output?

parsing the logs seems fragile and expensive to me...

Reply:
By using it we could even parse the output as json:

```
$ docker run --rm -it  istio/pilot:1.20.4 version
version.BuildInfo{Version:"1.20.4", GitRevision:"ff333a3512ba8321f40d90b6fc041db04fcb280b", GolangVersion:"go1.21.8", BuildStatus:"Clean", GitTag:"1.20.4"}
```

Just inspect the `.Version` field.

Reply:
perhaps you can even import this file to use this struct directly: https://github.com/istio/istio/blob/master/pkg/version/version.go#L37

Reply:
But we still have to use a regexp to extract the stable part of the version.